src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.configuration.AdminDNs;
+import org.opensearch.security.dlic.rest.support.Utils;
+import org.opensearch.security.privileges.PrivilegesEvaluator;
+import org.opensearch.security.securityconf.impl.v7.ActionGroupsV7;
+import org.opensearch.security.securityconf.impl.v7.RoleV7;
+import org.opensearch.security.support.WildcardMatcher;
+import org.opensearch.security.user.User;
+
+public class RestApiAdminPrivilegesEvaluator {
+
+    protected final Logger logger = LogManager.getLogger(RestApiAdminPrivilegesEvaluator.class);
+
+    private final static String REST_API_PERMISSION_PREFIX = "restapi:admin";
+
+    private final static String REST_ENDPOINT_PERMISSION_PATTERN =
+            REST_API_PERMISSION_PREFIX + "/%s";
+
+    private final static String REST_ENDPOINT_ACTION_PERMISSION_PATTERN =
+            REST_API_PERMISSION_PREFIX + "/%s/%s";
+
+    public final static WildcardMatcher REST_API_PERMISSION_PREFIX_MATCHER =
+            WildcardMatcher.from(REST_API_PERMISSION_PREFIX + "/*");
+
+    @FunctionalInterface
+    interface PermissionBuilder {
+        String build(final Endpoint endpoint, final String action);
+    }
+
+    public final static List<Endpoint> ENDPOINTS_WITH_PERMISSIONS =
+            ImmutableList.of(
+                    Endpoint.ACTIONGROUPS,
+                    Endpoint.ALLOWLIST,
+                    Endpoint.AUDIT,
+                    Endpoint.INTERNALUSERS,
+                    Endpoint.NODESDN,
+                    Endpoint.ROLES, Endpoint.ROLESMAPPING, Endpoint.TENANTS, Endpoint.SSL
+            );
+
+    public final static PermissionBuilder PERMISSION_BUILDER = (endpoint, action) -> {
+        switch (endpoint) {
+            case ACTIONGROUPS:
+                return buildEndpointPermission(Endpoint.ACTIONGROUPS);
+            case ALLOWLIST:
+                return buildEndpointPermission(Endpoint.ALLOWLIST);
+            case AUDIT:
+                return buildEndpointPermission(Endpoint.AUDIT);
+            case INTERNALUSERS:
+                return buildEndpointPermission(Endpoint.INTERNALUSERS);
+            case NODESDN:
+                return buildEndpointPermission(Endpoint.NODESDN);
+            case ROLES:
+                return buildEndpointPermission(Endpoint.ROLES);
+            case ROLESMAPPING:
+                return buildEndpointPermission(Endpoint.ROLESMAPPING);
+            case TENANTS:
+                return buildEndpointPermission(Endpoint.TENANTS);
+            case SSL:
+                switch (action) {
+                    case "certs":
+                        return buildEndpointActionPermission(Endpoint.SSL, "certs/info");
+                    case "reloadcerts":
```

Comment:
No sure that I got the question. If it is about `switch` and `break` than it is ok to use  `return` inside the `switch` block. If it is about 2 diff actions so the answer is this:
It was done this way since I merged functionality for SSL certs into 1 action. Before that there were 2 action handlers (2 classes) to keep backward compatibility I invented an action.

Reply:
No you are totally correct. I just misread the statement. Sorry about that.